server/plugin.go

```diff
@@ -774,9 +775,15 @@ func (p *Plugin) syncUsers() {
 				_, err := p.API.UpdateUser(mmUser)
 				if err != nil {
 					if err.Id == "app.user.save.username_exists.app_error" {
-						mmUser.Username += "-" + fmt.Sprint(userSuffixID)
-						userSuffixID++
-						continue
+						mmUser.Username = mmUsername               // assign the original username
+						username += "-" + fmt.Sprint(userSuffixID) // add the sufix before checking again
+						if mmUser.Username != username {
```

Comment:
I'm having trouble understanding this block, since it seems like it's guaranteed to always be true given the assignments prior?